### ISSUE TITLE
Use a LaTeX engine for static equations and figures generation

### ISSUE THREAD

**Issue by owiecc** [tlienart/Franklin.jl#945]
I would love to see a full-blown LaTeX support. Please take a look at this wonderful site: https://krasjet.com/voice/karasu/

The author wrote a custom pipeline in Haskell: `.md` with LaTeX mixins → LaTeX engine → [dvisvgm](https://dvisvgm.de/) → `.svg` → embed in `.html`. As a result, you can put any LaTeX code, including a custom preamble, and embed the result in a website. This would have the benefit of not needing any JS for math or LaTeX-derived figures (Tikz, CircuiTikz, chemfig, …). This also enables custom fonts to be embedded as one could e.g. use a XeLaTeX engine. 

There already is [TikzPictures.jl](https://github.com/JuliaTeX/TikzPictures.jl) (that does 90% of the heavy LaTeX lifting) and [LaTeX.jl](https://github.com/rened/LaTeX.jl) (which appears to be unmaintained).

**Comment by tlienart** [comment]
This won't happen inside Franklin. 

Now on the more constructive side, you can define your own plugins which can call arbitrary code including Haskell and Pandoc, so if you can write code that ends up generating the HTML you want; then it's trivial to have Franklin include the result. 

Additional note (similar to previous point) that you can already use things like TikzPictures, Latexify etc as well; there are demos for TikzCDs and Latexify here: https://franklinjl.org/demos/#009_custom_environment_for_tikzcd and here: https://franklinjl.org/demos/#004_use_latexifyjl. 

### Note

I'm not sure why you think this is a good idea by the way.

For equations KaTeX is excellent and can be pre-rendered to HTML (which is the case by default in Franklin when deploying the website). E.g. this page has equations and no js: https://tlienart2.github.io/#latex_and_maths

For figures, you can already use things like TikzPictures.jl, TikzCDs.jl, PGFPlots(X) etc as long as you set up your GitHub action runner properly since you need a tex engine, this is done in the demo pointed to earlier for TikzCD. 

So in conclusion:

- if you have a pipeline in any language that you'd like to use, you can
- there's already a path to not having JS dependencies for maths which you can use
- you can already use plotting/figures packages that generate SVG or PNG and embed that

**Comment by owiecc** [comment]
This is only mostly getting a high quality equations (e.g. custom fonts) without JS. 

I agree this would need to be done in an external package. I think I will have a chat with [JuliaTeX](https://github.com/JuliaTeX) to generalise the [TikzPictures.jl](https://github.com/JuliaTeX/TikzPictures.jl) package.

**Comment by tlienart** [comment]
Just a last note before completely closing this, 

* TikzPictures relies on luatex
* KaTeX relies on JS
* both can be pre-rendered, one to SVG, one to HTML

The only positive side of TikzPictures is that you can embed the font in the produced SVG which means you don't have to have users loading the fonts and have reduced CSS usage. Note that you can switch fonts in KaTeX too and pre-render with a different set.

These reasons might still somehow be sufficient on your side to push this further and, why not. 

One last note you might find useful: if the user ends up wanting to do things on CI, with a `luatex+pdf2svg` based system, you need to install the dependency on CI; something like

```
run: |
    sudo apt-get update -qq
    sudo apt install -y pdf2svg texlive-latex-base texlive-binaries texlive-pictures texlive-latex-extra texlive-luatex
```

 Currently this is also required for the likes of PGFPlotsX for instance. This adds build-time overhead of over 1 minute which may be irrelevant in your scenario though (especially because Tikz* packages are pretty slow in terms of ttfx); it might still be good to know.

Good luck!

**Comment by owiecc** [comment]
Thanks for the tips. I have a good base in the form of TikzPictures and [LaTeXiT](https://www.chachatelier.fr/latexit/) so maybe I will try to cook up something.
